src/Oscoin/Data/Core.hs

```diff
@@ -0,0 +1,502 @@
+-- | Core ledger data types.
+--
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fno-warn-redundant-constraints #-}
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
+module Oscoin.Data.Core where
```

Comment:
This is a very generic module name. What do think about `Oscoin.World` or `Oscoin.Ledger`

Reply:
Can we also have explicit exports to distinguish between internal and public methods?

Reply:
Indeed, maybe `Oscoin.Data.Ledger`?